pallets/xyk/src/lib.rs

```diff
@@ -503,25 +526,26 @@ impl<T: Trait> Module<T> {
 
         result
             .saturated_into::<u128>()
-            .saturated_into::<T::Balance>()
+            .saturated_into::<BalanceOf<T>>()
     }
 
     pub fn get_liquidity_asset(
-        first_asset_id: T::AssetId,
-        second_asset_id: T::AssetId,
-    ) -> T::AssetId {
+        first_asset_id: CurrencyIdOf<T>,
+        second_asset_id: CurrencyIdOf<T>,
+    ) -> CurrencyIdOf<T> {
+        //TODO get rid of unwraps
         if <LiquidityAssets<T>>::contains_key((first_asset_id, second_asset_id)) {
-            <LiquidityAssets<T>>::get((first_asset_id, second_asset_id))
+            <LiquidityAssets<T>>::get((first_asset_id, second_asset_id)).unwrap()
```

Comment:
I do not think we should use unwrap(), as get_liquidity_asset might get called without checking if LiquidityAssets contains key (either way)

Reply:
Also in some places to check if the pool exists Pools is used and in others LiquidityAssets is used, I'm not sure if that's the right approach.